### PR TITLE
Move unused classes from org.gradle.util to org.gradle.util.internal package 

### PR DESCRIPTION
**Pull request by donat**
This PR implements a first round of cleanup in the `org.gradle.util` package. It moves classes that not referenced by any external repositories are moved to the `org.gradle.util.internal` package. The classes not even used internally are deleted.

The external search was done with two tools: Github search ([example](https://github.com/search?q=%22import+org.gradle.util%22-repo%3Agradle%2Fgradle&type=Code)) and grep.app ([example](https://grep.app/search?q=org.gradle.util.GradleVersion)).

Note to the person merging the PR: please don't squash the PR when merging. It will be easier to revert individual changes if we accidentally break an external plugin.

Note to the person reviewing: The PR is rather large, but it's split into tiny individual commits. It's easier to do the review commit-by-commit rather than in one round.

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle

**Comment by donat** [comment]
@bot-gradle QuickFeedbackLinux

**Comment by donat** [comment]
@bot-gradle test QuickFeedbackLinux

**Comment by donat** [comment]
@bot-gradle test rFM

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -44,3 +44,29 @@ Some plugins will break with this new version of Gradle, for example because the
 The `JacocoMerge` task was used for merging coverage reports from different subprojects into a single report.
 The same functionality is also available on the `JacocoReport` task.
 Because of the duplication, `JacocoMerge` is now deprecated and scheduled for removal in Gradle 8.0.
+
+=== Potential breaking changes
+
+The following classes were moved from `org.gradle.util` to the `org.gradle.util` package:
```
Comment (rieske) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:50]: *to the `org.gradle.util.internal`

Comment (donat) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Fixed.

**Review by rieske — APPROVED** [review]

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle

**Comment by donat** [comment]
@bot-gradle test and merge this

**Comment by donat** [comment]
@bot-gradle test and merge this

**Pull request merged**